lux-lib/src/lockfile/mod.rs

```diff
@@ -500,7 +556,7 @@ impl LockfilePermissions for ReadOnly {}
 impl LockfilePermissions for ReadWrite {}
 
 #[derive(Clone, Debug, Serialize, Deserialize, Default)]
-pub(crate) struct LocalPackageLock {
+pub struct LocalPackageLock {
```

Comment:
suggestion: Instead of exposing this type, can we add delegate methods?

Reply:
Sure, I'll get to that near the end of this PR :+1:

Reply:
With the refactor we don't even need to delegate anything anymore, so I'll just revert these parts of the changes.